### PR TITLE
feat: Close search pop-up when escape is pressed

### LINKED ISSUE #418: Feature:  Search pop-up don't close If I shot Escape
### Type of feature

🍕 Feature

### Current behavior

image

### Suggested solution

I was expecting the pop-up to close when I pressed escape

#415

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by RozzaysRed**
Utilize headless ui combobox as command palette

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR replaces the existing input with headless ui's combobox and closes the search results when pressing esc.
## Related Tickets & Documents

Fixes #418 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/components/Hero.tsx
Comment (bdougie) [inline src/components/Hero.tsx:49]:
```suggestion
```
Remove extra space?

**Comment by TAKANOME-DEV** [comment]
@RozzaysRed [Headless ui combobox](https://headlessui.com/react/combobox) support keyboard navigation. Wondering why it's not working here :thinking:

File: src/components/Hero.tsx
Comment (NsdHSO) [inline src/components/Hero.tsx:76]:
```suggestion
        <div className="mt-11 px-4 gap-x-2.5 py-2.5  bg-white shadow-2xl rounded-2xl md:min-w-[26.375rem] flex">
```

Justify-between => justify-content: justify-between . It is good behavior, we don't want to change the base Frame.
Screenshot 2023-02-21 at 09 25 13

**Comment by RozzaysRed** [comment]
> @RozzaysRed [Headless ui combobox](https://headlessui.com/react/combobox) support keyboard navigation. Wondering why it's not working here :thinking: 

@TAKANOME-DEV i see what you me. Ill look in to why that is.

**Comment by bdougie** [comment]
Please provide a Desktop Screenshots/Recordings

File: src/components/Hero.tsx
Comment (NsdHSO) [reply src/components/Hero.tsx]: Could you please add `flex:1`

**Review by bdougie — APPROVED** [review]
Works for me. I did not quite understand if @takanome-dev was presenting a reason to block this, but if so I will hold off on merging for a bit.

**Comment by takanome-dev** [comment]
> Works for me. I did not quite understand if @takanome-dev was presenting a reason to block this, but if so I will hold off on merging for a bit.

It was about a11y. It's all good now :pizza:

**Comment by takanome-dev** [comment]
@RozzaysRed This pr can be merged after the conflicts are fixed :smiley:

**Pull request merged**
